### PR TITLE
fix(deps): Update dependencies (Kotlin 1.6)

### PR DESCRIPTION
**Pull request by Filmaluco**
Part of https://github.com/relaycorp/relayverse/issues/39

### CODE REVIEW

File: app/src/main/res/xml/data_extraction_rules.xml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+   Sample data extraction rules file; uncomment and customize as necessary.
+   See https://developer.android.com/about/versions/12/backup-restore#xml-changes
+   for details.
+-->
+<data-extraction-rules>
+    <device-transfer>
+        <include domain="sharedpref" path="pref_courier.xml"/>
```
Comment (sdsantos) [inline app/src/main/res/xml/data_extraction_rules.xml:9]: Couldn't it be * here as well?

**Review by sdsantos — APPROVED** [review]

File: app/build.gradle [hunk 2]
```diff
@@ -124,55 +123,55 @@ dependencies {
...
     // Local and Internet-based Parcel Delivery Connections (PDCs)
-    implementation 'tech.relaycorp:poweb:1.5.22'
+    implementation 'tech.relaycorp:poweb:1.5.25'
     implementation "io.ktor:ktor-server-core:$ktorVersion"
     implementation "io.ktor:ktor-server-netty:$ktorVersion"
     implementation "io.ktor:ktor-websockets:$ktorVersion"
     implementation "io.ktor:ktor-client-websockets:$ktorVersion"
     implementation "io.ktor:ktor-client-cio:$ktorVersion"
-    implementation "tech.relaycorp:doh:1.0.4"
+    implementation "tech.relaycorp:doh:1.0.4" // needs ktor update as well to avoid conflicts
```
Comment (gnarea) [inline app/build.gradle:159]: This kind of comments are best kept in the PR (as inline comments) as they're only relevant in this PR. Also in this particular case it's on the GH issue with the big TODO list.

File: app/build.gradle [hunk 3]
```diff
@@ -124,55 +123,55 @@ dependencies {
...
     implementation "androidx.room:room-ktx:$room_version"
 
     // required to avoid crash on Android 12 API 31
-    implementation 'androidx.work:work-runtime-ktx:2.7.0-beta01'
+    implementation 'androidx.work:work-runtime-ktx:2.7.1'
 
     // Preferences
     implementation 'androidx.preference:preference-ktx:1.1.1'
-    implementation 'com.fredporciuncula:flow-preferences:1.5.0'
+    implementation 'com.fredporciuncula:flow-preferences:1.5.0' // needs a higher version of kotlin
```
Comment (gnarea) [inline app/build.gradle:174]: Ditto.

File: app/lint.xml [hunk 4]
```diff
@@ -10,6 +10,9 @@
     <issue id="InvalidPackage">
         <!-- Ignore errors about BC importing javax.naming because we don't use those modules -->
         <ignore path="**/bcpkix-*.jar" />
+        <ignore path="**/org.jacoco/**" />
+        <ignore path="**/grpc-core*.jar" />
+        <ignore path="**/kotlinx-coroutines-core-jvm-1.5.2.jar" />
```
Comment (gnarea) [inline app/lint.xml:15]: Please include the reason for ignoring these packages (see Bouncy Castle entry above). This is partly to make sure we aren't silencing issues that we should actually be fixing, and partly to document the offending dependency (to remove the entry if the dependency is dropped in the future).

File: build.gradle [hunk 5]
```diff
@@ -4,27 +4,27 @@ buildscript {
...
     }
     repositories {
         google()
-        jcenter()
-        maven {
-            url "https://plugins.gradle.org/m2/"
-        }
+        mavenCentral()
+        maven { url "https://plugins.gradle.org/m2/" }
+        maven { url 'https://oss.sonatype.org/content/repositories/snapshots' }
```
Comment (gnarea) [inline build.gradle:13]: Why do we need snapshots in the production code?

**Review by gnarea — CHANGES_REQUESTED** [review]

File: app/build.gradle [hunk 6]
```diff
@@ -162,17 +161,17 @@ dependencies {
...
     implementation "androidx.room:room-ktx:$room_version"
 
     // required to avoid crash on Android 12 API 31
-    implementation 'androidx.work:work-runtime-ktx:2.7.0-beta01'
+    implementation 'androidx.work:work-runtime-ktx:2.7.1'
 
     // Preferences
     implementation 'androidx.preference:preference-ktx:1.1.1'
-    implementation 'com.fredporciuncula:flow-preferences:1.5.0'
+    implementation 'com.fredporciuncula:flow-preferences:1.5.0'
```
Comment (Filmaluco) [inline app/build.gradle:174]: // needs a higher version of kotlin

File: app/build.gradle [hunk 7]
```diff
@@ -124,34 +123,34 @@ dependencies {
...
+    implementation 'tech.relaycorp:cogrpc-okhttp:1.1.12'
     testImplementation "tech.relaycorp:awala-testing:$awalaTestingVersion"
     androidTestImplementation "tech.relaycorp:awala-testing:$awalaTestingVersion"
 
     // Android TLS support for Netty
-    implementation "io.netty:netty-handler:4.1.65.Final"
+    implementation "io.netty:netty-handler:4.1.77.Final"
     implementation 'org.conscrypt:conscrypt-android:2.5.2'
 
     // Local and Internet-based Parcel Delivery Connections (PDCs)
```
Comment (Filmaluco) [inline app/build.gradle:152]: `    implementation "tech.relaycorp:doh:1.0.4"` needs ktor update as well to avoid conflicts

**Review by gnarea — APPROVED** [review]
Obrigado!

**Pull request merged**